app/Http/Controllers/ArticleController.php

```diff
@@ -70,31 +61,69 @@ public function index()
+            $items = $articles->concat($highlights)->concat($experiences)->concat($videos)->sortByDesc('sort_date');
+
+            if (request('type')) {
+                $items = $items->where('type', (Str::singular(request('type'))));
+            }
+
+            $articlesCount = $items->count();
 
             $perPage = self::ARTICLES_PER_PAGE;
             $currentPage = LengthAwarePaginator::resolveCurrentPage();
-            $currentItems = $combined->slice(($currentPage - 1) * $perPage, $perPage)->all();
-            $paginator = new LengthAwarePaginator($currentItems, count($combined), $perPage, $currentPage, ['path' => LengthAwarePaginator::resolveCurrentPath()]);
+            $currentItems = $items->slice(($currentPage - 1) * $perPage, $perPage)->all();
+            $paginator = new LengthAwarePaginator($currentItems, count($items), $perPage, $currentPage, ['path' => LengthAwarePaginator::resolveCurrentPath()]);
 
             $articles = $paginator;
         } else {
             // Retrieve experiences entires
             $articles = Experience::webPublished()->articlePublished()->paginate(self::ARTICLES_PER_PAGE);
         }
 
+        $types = [
+            [
+                'label' => 'All',
+                'href' => route('articles', ['category' => request()->query('category')]),
+                'active' => empty(request()->all()),
+                'ajaxScrollTarget' => 'listing',
+            ],
+            [
+                'label' => 'Articles',
+                'href' => route('articles', array_merge(['type' => 'articles'], null !== request()->query('category') ? ['category' => request()->query('category')] : [])),
+                'active' => request()->get('type') == 'articles',
+                'ajaxScrollTarget' => 'listing',
+            ],
+            [
+                'label' => 'Highlights',
+                'href' => route('articles', array_merge(['type' => 'highlights'], null !== request()->query('category') ? ['category' => request()->query('category')] : [])),
+                'active' => request()->get('type') == 'highlights',
+                'ajaxScrollTarget' => 'listing',
+            ],
+            [
+                'label' => 'Videos',
+                'href' => route('articles', array_merge(['type' => 'videos'], null !== request()->query('category') ? ['category' => request()->query('category')] : [])),
+                'active' => request()->get('type') == 'videos',
+                'ajaxScrollTarget' => 'listing',
+            ],
+            [
+                'label' => 'Experiences',
+                'href' => route('articles', array_merge(['type' => 'experiences'], null !== request()->query('category') ? ['category' => request()->query('category')] : [])),
+                'active' => request()->get('type') == 'experiences',
+                'ajaxScrollTarget' => 'listing',
+            ]
+        ];
```

Comment:
You could probably reduce some redundancy with a foreach loop for these, something like:
```php
$types = [
    [
      'label' => 'All',
      'href' => route('articles', ['category' => request()->query('category')]),
      'active' => empty(request()->all()),
      'ajaxScrollTarget' => 'listing',
    ],
];
foreach(['Articles', 'Highlights', 'Videos', 'Experiences'] as $label) {
  $types[] = [
    'label' => $label,
    'href' => route(strtolower($label), array_filter(['type' => strtolower($label), 'category' => $category])),
    'active' => $type == strtolower($label),
    'ajaxScrollTarget' => 'listing',
  ];
}
```